packages/datatrak-web/src/routes/PrivateRoute.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { ReactNode } from 'react';
+import { Navigate, Outlet, useLocation } from 'react-router-dom';
+import { useCurrentUser } from '../api';
+import { ROUTES } from '../constants';
+
+// Reusable wrapper to handle redirecting to login if user is not logged in and the route is private
+export const PrivateRoute = ({ children }: { children?: ReactNode }): any => {
```

Comment:
This is split our from the routes file since that was getting quite big and then it could be simplified slightly since we don't have to handle the loading state anymore